torchrl/envs/vec_env.py

```diff
@@ -624,14 +634,17 @@ def _reset(self, tensordict: TensorDictBase, **kwargs) -> TensorDictBase:
                 continue
             _td = _env._reset(tensordict=tensordict_, **kwargs)
             self.shared_tensordicts[i].update_(
-                _td.select(*self._selected_keys, strict=False)
+                _td.select(*self._selected_reset_keys, strict=False)
             )
 
         if self._single_task:
             # select + clone creates 2 tds, but we can create one only
-            out = TensorDict({}, batch_size=self.shared_tensordict_parent.shape)
+            out = TensorDict(
+                {}, batch_size=self.shared_tensordict_parent.shape, device=self.device
+            )
             for key in self._selected_reset_keys:
-                _set_single_key(self.shared_tensordict_parent, out, key, clone=True)
+                if _unravel_key_to_tuple(key)[-1] != "_reset":
```

Comment:
again, no need to check here if you just exclude the reset key from selected reset keys

Reply:
but if you wanna check, there could be multiple reset keys